bundle/src/main/java/com/meltmedia/dropwizard/etcd/cluster/ClusterAssignmentService.java

```diff
@@ -202,18 +230,37 @@ public void startNodeAssignmentTask() {
         () -> {
           assignmentTask.mark();
           try {
-            int processCount = thisProcessCount.get();
-            int currentNodeCount = stateTracker.getState().memberCount();
-            int total = totalProcessCount.get();
+            ProcessorState processorState = this.processorState.get();
+            AssignmentState assignmentState = this.assignmentState.get();
+            State clusterState = stateTracker.getState();
+            long lastSeenIndex = Math.max(assignmentState.etcdIndex, clusterState.lastModifiedIndex());
+            
+            if( lastAssignmentIndex > lastSeenIndex ) return;
+            boolean active = clusterState.hasMember(thisNode.getId()) && processorState.hasProcessor(this.getId()) && running;
+            int processCount = assignmentState.nodeProcessCount();
+            int unassignedProcessCount = assignmentState.unassignedProcessCount();
+            int currentNodeCount = processorState.processorCount();
+            int total = assignmentState.totalProcessCount();
 
             int maxProcessCount =
-              currentNodeCount == 0 ? 0 : IntMath.divide(total, currentNodeCount,
+              currentNodeCount == 0 || !active ? 0 : IntMath.divide(total, currentNodeCount,
                 RoundingMode.CEILING);
 
-            if (processCount < maxProcessCount && !unassigned.isEmpty()) {
-              for (String toAssign : unassigned) {
+            if( !active && processCount == 0 ) {
+                if( emptyLock.tryLock() ) {
+                  try {
+                    empty.signalAll();
+                  }
+                  finally {
+                    emptyLock.unlock();
+                  }
+                }
```

Comment:
This block should be wrapped up with the other lock code into a simple shutdown latch.